vendors/halo2/README.md

```diff
@@ -0,0 +1,137 @@
+
+   ``` diff
+   # In your Cargo.toml
+   - halo2_proofs = { git = "https://github.com/zcash/halo2.git" }
+   + halo2_proofs = { git = "https://github.com/kroma-network/halo2.git", branch = "halo2-with-tachyon" }
+   ```
+
+2. Install the tachyon debian package, See *Debian packaging* section in [how_to_build.md](../how_to_build.md).
+
+### Migration Steps
+
+1. Create Proving Key for Tachyon
+
+    ``` diff
+      use halo2_proofs::{
+    +   bn254::ProvingKey as TachyonProvingKey,
+        halo2curves::bn256::{Bn256, Fr},
+        plonk::keygen_pk2,
+        poly::kzg::commitment::ParamsKZG,
+      };
+
+      // Use just your `circuit` and `instances` so.
+      let circuit = YourCircuit<_>;
+      let instances = <YourInstances>
+
+      // `k`: log of degree, `s`: toxic-waste secret.
+      let k: u32 = 5;
+      let s = Fr::from(2);
+
+      // Generate `params`.
+      let params = ParamsKZG<Bn256>::unsafe_setup_with_s(k, s);
+
+      // Generate `pk` as proving key.
+      let pk = keygen_pk2(&params, &circuit).expect("keygen_pk2 should not fail");
+    + let mut tachyon_pk = {
+    +     // Please remember to release `pk_bytes` immediately after
+    +     // deserialization, as it can occupy big memory.
+    +     let mut pk_bytes: Vec<u8> = vec![];
+    +     pk.write(&mut pk_bytes, halo2_proofs::SerdeFormat::RawBytesUnchecked)
+    +         .unwrap();
+    +     drop(pk);
+    +     TachyonProvingKey::from(pk_bytes.as_slice())
+    + };
+   ```
+
+2. Change transcript for Tachyon
+
+    ``` diff
+    -  use halo2_proofs::transcript::{
+    -     halo2curves::bn256::G1Affine,
+    -     Challenge255, PoseidonWrite,
+    - };
+    + use halo2_proofs::bn254::PoseidonWrite as TachyonPoseidonWrite;
+
+    - let mut transcript = PoseidonWrite::<_, G1Affine, Challenge255<_>>::init(vec![]);
+    + let mut transcript = TachyonPoseidonWrite::init(vec![]);
+    ```
+
+    Tachyon supports *PoseidonWrite*, *Blake2bWrite*, and *Sha256Write* as transcript methods.
+
+3. Change RNG (Random Number Generator) for Tachyon
+
+    ``` diff
+    - use rand_core::{RngCore, SeedableRng};
+    - use rand_xorshift::XorShiftRng;
+    + use halo2_proofs::xor_shift_rng::XORShiftRng;
+    + use rand_core::SeedableRng;
+
+      let seed: [u8; 16] = <SpecificSeed>;
+    - let rng = rand_xorshift::XorShiftRng::from_seed(seed);
+    + let rng = XORShiftRng::from_seed(seed);
+    ```
+
+4. Create Prover for Tachyon
+
+    ``` diff
+    + use halo2_proofs::{
+    +     bn254::{
+    +         GWCProver as TachyonGWCProver,
+    +         TachyonProver,
+    +     },
+    +     consts::TranscriptType,
+    +     halo2curves::bn256::Bn256,
+    +     poly::kzg::commitment::KZGCommitmentScheme,
+    + }
+    +
+    + let mut prover = {
+    +     let mut params_bytes = vec![];
+    +     // Please remember to release `params_bytes` immediately after
+    +     // deserialization, as it can occupy big memory.
```

Comment:
```suggestion
    +     // Please remember to release `params` and `params_bytes` immediately after
    +     // deserialization, as it can occupy big memory.
```